src/extractor/guidance/intersection_generator.cpp

```diff
@@ -577,9 +580,13 @@ IntersectionGenerator::GetActualNextIntersection(const NodeID starting_node,
     // to prevent endless loops
     const auto termination_node = node_based_graph.GetTarget(via_edge);
 
-    while (result.size() == 2 &&
-           node_based_graph.GetEdgeData(via_edge).IsCompatibleTo(
-               node_based_graph.GetEdgeData(result[1].turn.eid)))
+    // using a maximum lookahead, we make sure not to end up in some form of loop
+    const constexpr std::size_t max_lookahead = 10;
+    for (std::size_t lookahead = 0;
+         lookahead < max_lookahead && (result.size() == 2 &&
+                                       node_based_graph.GetEdgeData(via_edge).IsCompatibleTo(
+                                           node_based_graph.GetEdgeData(result[1].turn.eid)));
+         ++lookahead)
```

Comment:
For my own understanding: you try to find the first compatible edge but now with a limit so that in edge cases (as the one triggering this) we only end up looking at a limited amount (10 here) to prevent loops. Why the 10 here instead of e.g. storing and checking for a actual loop?